pkg/util/platformworkloadidentity/federatedcredentials.go

```diff
@@ -18,7 +18,7 @@ const (
 )
 
 func GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, identityResourceId azure.Resource, serviceAccountName string) string {
-	clusterResourceKey := fmt.Sprintf("%s-%s-%s", clusterResourceId.SubscriptionID, clusterResourceId.ResourceGroup, clusterResourceId.ResourceName)
+	clusterResourceKey := fmt.Sprintf("%s_%s", serviceAccountName, clusterResourceId.ResourceName)
```

Comment:
We'll need to replace any occurrences of `:` in the serviceAccountName as well, `:` is not a valid character for federated credential names. We may want to consider parsing the name and trimming off the first two sections, as these service account names will generally start with `system:serviceaccount:` which is redundant for the purposes of naming the fedcred.